### PR TITLE
KT - added groceryIndex page and test

### PR DESCRIPTION
**Pull request by kkwen123**
In this PR, I added the GroceryIndexPage.js and GroceryIndexPage.test.js files, and made sure all of the corresponding tests passes.

### CODE REVIEW

File: frontend/src/tests/pages/Groceries/GroceryIndexPage.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <GroceryIndexPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        const name = screen.getByText("Sandwich");
+        expect(name).toBeInTheDocument();
+
```
Comment (BrennaScholte) [inline frontend/src/tests/pages/Groceries/GroceryIndexPage.test.js:93]: add price here

File: frontend/src/tests/pages/Groceries/GroceryIndexPage.test.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            </QueryClientProvider>
+        );
+
+        const createGroceryButton = screen.getByText("Create Grocery");
+        expect(createGroceryButton).toBeInTheDocument();
+        expect(createGroceryButton).toHaveAttribute("style", "float: right;");
+
+        const name = screen.getByText("Sandwich");
+        expect(name).toBeInTheDocument();
+
```
Comment (BrennaScholte) [inline frontend/src/tests/pages/Groceries/GroceryIndexPage.test.js:67]: add price here

**Review by BrennaScholte — DISMISSED** [review]
Please see comments, just need to add price field in two places!

**Review by ritvibhatt — APPROVED** [review]
LGTM

**Pull request merged**